policybot/pkg/resultgatherer/resultgatherer.go

```diff
@@ -447,39 +488,59 @@ func (trg *TestResultGatherer) GetPostSubmitTestResult(ctx context.Context, test
 	testResult.RunPath = testRun
 	testResult.Done = false
 
-	records, err := trg.getInformationFromCloneFile(ctx, testRun)
+	pj, err := trg.getInformationFromProwFile(ctx, testRun)
 	if err != nil {
-		return
+		return nil, err
 	}
 
-	if len(records) < 1 {
-		return nil, fmt.Errorf("test %s %s has an empty clone file.  Cannot proceed", testName, testRun)
+	if pj.Status.State == TriggeredState || pj.Status.State == PendingState {
+		return nil, fmt.Errorf("test is still in progress")
 	}
-	record := records[0]
-
-	testResult.Sha, err = hex.DecodeString(record.FinalSha)
-	if err != nil {
-		return
+	if pj.Status.State == ErrorState {
+		testResult.StartTime = pj.Status.StartTime.Time
+		testResult.FinishTime = pj.Status.CompletionTime.Time
+		testResult.Result = "ERROR"
+	}
+	if pj.Status.State == AbortedState {
+		testResult.StartTime = pj.Status.StartTime.Time
+		testResult.FinishTime = pj.Status.CompletionTime.Time
+		testResult.Result = "ABORTED"
 	}
+	if pj.Status.State == SuccessState || pj.Status.State == FailureState {
+		records, err := trg.getInformationFromCloneFile(ctx, testRun)
+		if err != nil {
+			return nil, err
+		}
 
-	testResult.BaseSha = record.Refs.BaseSha
-	testResult.CloneFailed = record.Failed
+		if len(records) < 1 {
```

Comment:
Seems like this part of logic is similar with that in getInformationFromCloneFile, can possibly moved to common function?

Reply:
Chatted offline.  Difficult to consolidate this logic due to minor differences.